### PR TITLE
Bug 1907353: Remove OVS daemonsets

### PR DESCRIPTION
**Pull request by juanluisvaladas**
In 4.6 we moved OVS to systemd, we needed to have a pod capable of
running OVS during the upgrade, but after that it became unnecessary.

In 4.7 we're guaranteed to launch ovs in systemd, both when installed
from scratch and when upgrading from 4.6, therefore there is no use for
the OVS pod other than being able to gather the logs from the apiserver,
we cannot even restart the OVS daemons by deleting the pod, wihch is
confusing.

### CODE REVIEW

**Comment by juanluisvaladas** [comment]
@squeed @danwinship @trozet @abhat @pliurh
Do you agree with removing this? I think it's removing unnecessary complexity and possible points of failure.

**Comment by abhat** [comment]
Do we need to keep the pod to be able to get to the ovs logs?

**Comment by juanluisvaladas** [comment]
> Do we need to keep the pod to be able to get to the ovs logs?

I don't think we _need_ it. The logs are still present on the journal, and can be read with journalctl.

If we really want the logs to be readable from the apiserver, we should at least remove all the logic and just leave the tail.

**Comment by squeed** [comment]
If you make sure that must-gather gets openvswitch logs, then I'll approve this :-)

**Review by danwinship — COMMENTED** [review]
I agree it's not worth keeping these around just to provide easy access to the logs

**Comment by juanluisvaladas** [comment]
@squeed 

```
$ cat sos_commands/openvswitch/journalctl_--no-pager_--unit_openvswitch
-- Logs begin at Sat 2020-11-28 21:21:53 UTC, end at Sun 2020-11-29 20:04:17 UTC. --
Nov 29 02:02:56 localhost systemd[1]: Dependency failed for Open vSwitch.
Nov 29 02:02:56 localhost systemd[1]: openvswitch.service: Job openvswitch.service/start failed with result 'dependency'.

$ cat etc/redhat-release 
Red Hat Enterprise Linux CoreOS release 4.6
```

It's not in must gather but it's included in a sosreport.

File: bindata/network/openshift-sdn/sdn-ovs.yaml [hunk 1]
```diff
@@ -1,215 +0,0 @@
-{{if .InstallOVS}}
```
[the remark this answers is not in the document]
Comment (juanluisvaladas) [reply bindata/network/openshift-sdn/sdn-ovs.yaml:1]: Removed it and created this PR: https://github.com/openshift/api/pull/802

**Comment by squeed** [comment]
Looks like you have to update a few unit tests, then I'll lgtm.

/approve

**Comment by juanluisvaladas** [comment]
> Looks like you have to update a few unit tests, then I'll lgtm.
D'oh! Fixed.

```
[jdesousa@systemv /home/jdesousa/go/src/github.com/openshift/cluster-network-operator] (remove-ovs-pod)
$ hack/test-go.sh 
?   	github.com/openshift/cluster-network-operator/cmd/cluster-network-operator	[no test files]
?   	github.com/openshift/cluster-network-operator/cmd/cluster-network-renderer	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/apis/network/v1	[no test files]
ok  	github.com/openshift/cluster-network-operator/pkg/apply	(cached)
?   	github.com/openshift/cluster-network-operator/pkg/bootstrap	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller/clusterconfig	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller/configmap_ca_injector	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller/operconfig	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller/pki	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/controller/proxyconfig	[no test files]
ok  	github.com/openshift/cluster-network-operator/pkg/controller/statusmanager	(cached)
?   	github.com/openshift/cluster-network-operator/pkg/names	[no test files]
ok  	github.com/openshift/cluster-network-operator/pkg/network	0.281s
?   	github.com/openshift/cluster-network-operator/pkg/platform/openstack	[no test files]
?   	github.com/openshift/cluster-network-operator/pkg/platform/openstack/util/cert	[no test files]
ok  	github.com/openshift/cluster-network-operator/pkg/render	(cached)
ok  	github.com/openshift/cluster-network-operator/pkg/util/ip	(cached)
ok  	github.com/openshift/cluster-network-operator/pkg/util/k8s	(cached)
ok  	github.com/openshift/cluster-network-operator/pkg/util/proxyconfig	(cached)
ok  	github.com/openshift/cluster-network-operator/pkg/util/validation	(cached)

```

**Comment by juanluisvaladas** [comment]
/hold
This is technically a feature by our definition so I guess we have to hold this until 4.8...

**Comment by danwinship** [comment]
>  This is technically a feature by our definition so I guess we have to hold this until 4.8...

Maybe the change in OVS logging behavior is a feature, but we should at least get rid of the code for actually launching the OVS daemons and leave just the `tail -f` code. As of 4.7, the "starting ovs" code is 100% a liability; in a correctly-functioning cluster it will never be used, but in some edge cases it might get triggered, which will most likely break the cluster.

**Comment by giannisalinetti** [comment]
I all, 

> @squeed
> 
> ```
> $ cat sos_commands/openvswitch/journalctl_--no-pager_--unit_openvswitch
> -- Logs begin at Sat 2020-11-28 21:21:53 UTC, end at Sun 2020-11-29 20:04:17 UTC. --
> Nov 29 02:02:56 localhost systemd[1]: Dependency failed for Open vSwitch.
> Nov 29 02:02:56 localhost systemd[1]: openvswitch.service: Job openvswitch.service/start failed with result 'dependency'.
> 
> $ cat etc/redhat-release 
> Red Hat Enterprise Linux CoreOS release 4.6
> ```
> 
> It's not in must gather but it's included in a sosreport.

Hi all, users have a great advantage in collecting logs directly from the ovs pod. The must-gather could be a nice alternative but don't you think that forcing users to access the node to collect journalctl logs is going against the containerized controle plane approach we pursued until now?

This will be probably felt like a regression from users. Please reconsider the must-gather option at least.

**Comment by squeed** [comment]
@giannisalinetti I hear what you're saying. However, there are other system services that only log to journald - e.g. the kubelet. In order to make this easy for users to diagnose, there is a shortcut for this in `oc`.

`oc adm node-logs --unit=openvswitch <node name>` is a shortcut for reading node service logs. This is what must-gather does, by the way.

So yes, this does change how we access debug logs, but it brings it in line with the rest of the system, rather than treating openvswitch as special.

**Comment by juanluisvaladas** [comment]
Hi, must-gather now collects this information since 4.7. Everything's ready to merge on 4.8
/hold cancel

**Comment by danwinship** [comment]
@juanluisvaladas can you remove the references to `OVS_LOG_LEVEL` in `README.md` and `bindata/network/ovn-kubernetes/ipsec.yaml`? (It appears that it never actually did anything in the latter.)

**Comment by danwinship** [comment]
(also, needs rebase)

**Comment by jluhrsen** [comment]
> (also, needs rebase)

I couldn't figure out how to personall update this PR directly, so created a new one with the rebase and removal of OVS_LOG_LEVEL:

https://github.com/openshift/cluster-network-operator/pull/1028

I figured it was quick and easy and not sure if @juanluisvaladas would have time.

**Comment by trozet** [comment]
deprecated in favor of #1028

**Pull request closed without merging**